ingester/src/data.rs

```diff
@@ -646,21 +633,29 @@ impl TableData {
     #[cfg(test)]
     pub fn new_for_test(
         table_id: TableId,
-        parquet_max_sequence_number: Option<SequenceNumber>,
         tombstone_max_sequence_number: Option<SequenceNumber>,
         partitions: BTreeMap<String, PartitionData>,
     ) -> Self {
         Self {
             table_id,
-            parquet_max_sequence_number,
             tombstone_max_sequence_number,
             partition_data: partitions,
         }
     }
 
     /// Return parquet_max_sequence_number
     pub fn parquet_max_sequence_number(&self) -> Option<SequenceNumber> {
-        self.parquet_max_sequence_number
+        let mut max: Option<SequenceNumber> = None;
+        for p in self.partition_data.values() {
+            if let Some(m) = p.data.max_persisted_sequence_number {
+                max = match max {
+                    Some(s) => Some(s.max(m)),
+                    None => Some(m),
+                };
+            }
+        }
+
+        max
```

Comment:
This can be simplified quite nicely:

```suggestion
        self.partition_data
            .values()
            .map(|p| p.data.max_persisted_sequence_number)
            .max()
            .flatten()
```

This works because all [`None` are less than a `Some`](https://doc.rust-lang.org/std/option/#comparison-operators) (`None < Some(0)` is true) so we can call `max()` on the iterator knowing that we'll only get a `None` if there are no `Some`, and the `Some` values will be maxed 👌